pkg/providers/imagefamily/resolver.go

```diff
@@ -99,8 +101,12 @@ func (r *defaultResolver) Resolve(
 	if err != nil {
 		return nil, err
 	}
+	kubernetesVersion, err := nodeClass.GetKubernetesVersion()
+	if err != nil {
+		return nil, err
+	}
 
-	imageFamily := getImageFamily(nodeClass.Spec.ImageFamily, staticParameters)
+	imageFamily := getImageFamily(nodeClass.Spec.ImageFamily, kubernetesVersion, staticParameters)
```

Comment:
I think this would work, but I don't loving having the added dependency on the k8s version here. 

imageFamily is just being used to map to the distro on the provisioning side of things. Not necessarily critical to get in before this, but the area could use some cleanup, and would prefer to get that done before expanding on some of the mucky code already.

Reply:
Agree that this area could use some cleanup, but the added dependency on Kubernetes version just reflects the reality that, for Azure Linux at least, the distro explicitly depends on Kubernetes version, using a different one from 1.32 on.